### ISSUE TITLE
Add datatypes for network addresses

### ISSUE THREAD

**Issue by psubotic** [souffle-lang/souffle#29]
We need some datatypes and associated operations for network addresses. I anticipate we will need 
some bit-vector logic here or compile it into some underlying data structure e.g., tries.

**Comment by HerbertJordan** [comment]
What kind of network addresses? Are you referring to IP addresses?

**Comment by psubotic** [comment]
yes

**Comment by HerbertJordan** [comment]
Shouldn't be too difficult. What operations are you envisioning?
- checks for sub-net memberships?
- computation of broadcast-addresses?

**Comment by b-scholz** [comment]
Is this an extension as proposed by 

http://research.microsoft.com/apps/pubs/default.aspx?id=215431

**Comment by b-scholz** [comment]
This has been resolved with the latest enhancement request - is this correct Paul?

**Comment by psubotic** [comment]
yes, lets assume so  unless someone explicitly asks for something more
advanced.
